hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -8853,16 +8849,12 @@ public void logAuditEvent(boolean succeeded, String userName,
     }
 
     public void logAuditMessage(String message) {
-      auditLog.info(message);
+      AUDIT_LOG.info(message);
     }
   }
 
   private static void enableAsyncAuditLog(Configuration conf) {
-    if (!(auditLog instanceof Log4JLogger)) {
```

Comment:
this is a key diff in the patch. today if the backend isn't log4j things should work, just not do async logging (remember, logback doesn't need this...), so this is possibly a regression

Reply:
I agree for this patch, but we do have sub-tasks coming up that would:

- Move dynamic log4j appenders to log4j.properties and let users update log4j.properties rather than let the code use log4j appenders
- Replace LOG4J references from production code to SLF4J

Reply:
Also, if downstream application excludes log4j references, hdfs would not work because in order for hdfs to add dynamic appenders, it needs log4j in the classpath at least.
And these references are on the server side only.